crates/npmrc/src/custom_deserializer.rs

```diff
@@ -118,11 +120,29 @@ mod tests {
         env::remove_var("PACQUET_HOME");
     }
 
+    #[test]
+    fn test_default_store_dir_with_pac_env_with_prefix() {
+        env::set_var("PACQUET_HOME", "~/prefix/pacquet_home");
+        let home_dir = dirs::home_dir().unwrap();
+        let store_dir = default_store_dir();
+        assert_eq!(store_dir, home_dir.join("prefix/pacquet_home/store"));
+        env::remove_var("PACQUET_HOME");
+    }
+
     #[test]
     fn test_default_store_dir_with_xdg_env() {
         env::set_var("XDG_DATA_HOME", "/tmp/xdg_data_home");
         let store_dir = default_store_dir();
         assert_eq!(store_dir, Path::new("/tmp/xdg_data_home/pacquet/store"));
         env::remove_var("XDG_DATA_HOME");
     }
+
+    #[test]
+    fn test_default_store_dir_with_xdg_env_with_prefix() {
+        env::set_var("XDG_DATA_HOME", "~/prefix/xdg_data_home");
+        let home_dir = dirs::home_dir().unwrap();
+        let store_dir = default_store_dir();
```

Comment:
Convert this to passing as arguments too.

Reply:
Same here

Reply:
https://github.com/pnpm/pacquet/pull/84#discussion_r1299117531